pallets/crowdloan-rewards/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+			T::RewardCurrency::deposit_creating(&payee, payable_amount);
+
+			// Emit event
+			Self::deposit_event(Event::RewardsPaid(payee, payable_amount));
+
+			Ok(Default::default())
+		}
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// User trying to claim an award did not have an claim associated with it. This may mean
+		/// they did not contribute to the crowdloan, or they have not yet associated a native id
+		/// with their contribution
+		NoAssociatedClaim,
+	}
+
+	#[pallet::storage]
+	pub type AccountsPayable<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, RewardInfo<T>>;
+
+	#[pallet::storage]
+	pub type UnassociatedContributions<T: Config> = StorageMap<_, Blake2_128Concat, T::RelayChainAccountId, RewardInfo<T>>;
+
+	// Design decision:
+	// Genesis config contributions are specified in relay-chain currency
+	// Conversion to reward currency happens when constructing genesis
+	// This pallets storages are all in terms of reward currency
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config> {
+		/// Contributions that have a native account id associated already.
+		pub associated: Vec<(T::AccountId, u32)>,
+		/// Contributions that will need a native account id to be associated through an extrinsic.
+		pub unassociated: Vec<(T::RelayChainAccountId, u32)>,
+		/// The ratio of (reward tokens to be paid) / (relay chain funds contributed)
+		/// This is dead stupid simple using a u32. So the reward amount has to be an integer
+		/// multiple of the contribution amount. A better fixed-ratio solution would be
+		/// https://crates.parity.io/sp_arithmetic/fixed_point/struct.FixedU128.html
+		/// We could also do something fancy and non-linear if the need arises.
+		pub reward_ratio: u32,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config> Default for GenesisConfig<T> {
+		fn default() -> Self {
+			Self {
+				associated: Vec::new(),
+				unassociated: Vec::new(),
+				reward_ratio: 1,
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
+		fn build(&self) {
+
+			// Warn if no contributions (associated or not) are specified
+			if self.associated.is_empty() && self.unassociated.is_empty() {
+				warn!("Rewards: No contributions configured. Pallet will not be useable.")
+			}
+
+			// Initialize storage for associated contributions
+			self.associated.iter().for_each(|(native_account, contrib)| {
+				let reward_info = RewardInfo{
+					total_reward: BalanceOf::<T>::from(*contrib) * BalanceOf::<T>::from(self.reward_ratio), //TODO safe math?
+					last_paid: 0.into(),
+				};
+				AccountsPayable::<T>::insert(native_account, reward_info);
+			});
+
+			// Initialize storage for UN-associated contributions
+			self.unassociated.iter().for_each(|(relay_account, contrib)| {
+				//TODO: 📠🍝
+				let reward_info = RewardInfo{
+					total_reward: BalanceOf::<T>::from(*contrib) * BalanceOf::<T>::from(self.reward_ratio), //TODO safe math?
```

Comment:
@4meta5 This is the conversion. It's super janky right now for sure and will be improved if we end up using this. But don't worry it is in there somewhere.